pkgs/jni/lib/src/jni.dart

```diff
@@ -62,12 +62,17 @@ abstract final class Jni {
     _dylibDir = dylibDir;
   }
 
+  static bool _initialized = false;
```

Comment:
This is a boolean per isolate.

The initialization only has to happen once.
I guess it doesn't hurt to do it multiple times, because the operation should be idempotent. (`runtime/include/dart_api_dl.c` updates function pointer addresses in a loop but should update them to the same value for subsequent runs.)

Please add a test that runs the initialization on two isolates to verify everything works as expected.